l1-contracts/src/core/Rollup.sol

```diff
@@ -167,14 +246,41 @@ contract Rollup is Leonidas, IRollup {
       revert Errors.Rollup__InvalidProof();
     }
 
+    blocks[header.globalVariables.blockNumber].isProven = true;
+
+    _progressState();
+
     emit L2ProofVerified(header.globalVariables.blockNumber, _proverId);
   }
 
-  function _computePublicInputHash(bytes calldata _header, bytes32 _archive)
-    internal
-    pure
-    returns (bytes32)
-  {
-    return Hash.sha256ToField(bytes.concat(_header, _archive));
+  /**
+   * @notice  Progresses the state of the proven chain as far as possible
+   *
+   * @dev     Emits `ProgressedState` if the state is progressed
+   *
+   * @dev     Will continue along the pending chain as long as the blocks are proven
+   *          stops at the first unproven block.
+   *
+   * @dev     Have a potentially unbounded gas usage. @todo Will need a bounded version, such that it cannot be
```

Comment:
You're saying in the event the pending chain grows too far beyond the proven chain? Wouldn't a timeliness requirement effectively bound this?

Reply:
Even with a timeliness, it "could" grow very large if blocks are coming in at max speed, and you could even have that it is past the timeliness but you just keep on extending and then at before someone is actually pruning it you would execute this. Sure, it is very unlikely, but you don't want to end up with a frozen system anyway.